### ISSUE TITLE
RetrieveModelMixin in the inherited ViewSet cannot work

### ISSUE THREAD

**Issue by lyn5466** [InterSIS/django-rest-encrypted-lookup#11]
Hi, 

I have a problem when working with this package. My viewset and url settings are shown as below.

**views.py**

```
class ActivityLogViewSet(viewsets.mixins.CreateModelMixin,
                         viewsets.mixins.ListModelMixin,
                         viewsets.mixins.RetrieveModelMixin,
                         EncryptedLookupGenericViewSet):
    """
    API endpoint that allows activity logs to be viewed or recorded.
    """
    queryset = ActivityLog.objects.all()
    serializer_class = ActivityLogSerializer
    lookup_field = 'id'
```

**urls.py**

```
router = routers.SimpleRouter()
router.register(r'activitylogs', views.ActivityLogViewSet)

urlpatterns = [
    url(r'^admin/', include(admin.site.urls)),
    url(r'^', include(router.urls)),
]
```

When my server start running, it did not provide a retrieve function.
Here is my query in browser: `http://127.0.0.1:8000/activitylogs/vq6dtebgu3ipafvkcsuwa4z3wi` where `vq6dtebgu3ipafvkcsuwa4z3wi` is the encrypted pk.
Then, I get errors as below:

```
Request Method: GET
Request URL:    http://127.0.0.1:8000/activitylogs/vq6dtebgu3ipafvkcsuwa4z3wi/
Django Version: 1.8.4
Exception Type: AttributeError
Exception Value:    
'ActivityLogViewSet' object has no attribute 'request'
Exception Location: /Applications/anaconda/lib/python2.7/site-packages/rest_framework/generics.py in get_serializer_context, line 77
Python Executable:  /Applications/anaconda/bin/python
Python Version: 2.7.9
Python Path:    
['/Users/mtk09566/Documents/project/logserver',
 '/Applications/anaconda/lib/python27.zip',
 '/Applications/anaconda/lib/python2.7',
 '/Applications/anaconda/lib/python2.7/plat-darwin',
 '/Applications/anaconda/lib/python2.7/plat-mac',
 '/Applications/anaconda/lib/python2.7/plat-mac/lib-scriptpackages',
 '/Applications/anaconda/lib/python2.7/lib-tk',
 '/Applications/anaconda/lib/python2.7/lib-old',
 '/Applications/anaconda/lib/python2.7/lib-dynload',
 '/Applications/anaconda/lib/python2.7/site-packages/cryptography-0.8-py2.7-macosx-10.5-x86_64.egg',
 '/Applications/anaconda/lib/python2.7/site-packages/setuptools-14.3-py2.7.egg',
 '/Applications/anaconda/lib/python2.7/site-packages',
 '/Applications/anaconda/lib/python2.7/site-packages/Sphinx-1.2.3-py2.7.egg',
 '/Applications/anaconda/lib/python2.7/site-packages/aeosa']
```

Hope you can help me solve this problem.
Thanks!

**Comment by JASchilz** [comment]
Acknowledged! Will take a look this weekend.

**Comment by JASchilz** [comment]
One idea and one question:

In ActivityLogViewset, try moving EncryptedLookupGenericViewSet to be the class's first parent. For example:

```
class ActivityLogViewSet(EncryptedLookupGenericViewSet,
                         viewsets.mixins.CreateModelMixin,
                         viewsets.mixins.ListModelMixin,
                         viewsets.mixins.RetrieveModelMixin):
```

I doubt that this will help, however I haven't worked with these packages in a while; maybe it's the solution.

Also, what is your version of django-rest-framework? Given that, I can try to replicate the issue with more confidence. I haven't used 3.2.x, but I would like to extend support to it.

**Comment by lyn5466** [comment]
Hi, 

Much thanks for your reply!
- First, by changing the ActivityLogViewSet as the class's first parent, I still cannot have the retrieve function, and the program keeps yelling:

```
Exception Type: AttributeError
Exception Value: 'ActivityLogViewSet' object has no attribute 'request'
```
- Second, my version of django-rest-framework is 3.0.5.

More details for reference:
- There is a log in my database with `id = 23`
- When using my browser to connect to `127.0.0.1:8000/activitylogs/`, I see the log with `id = vq6dtebgu3ipafvkcsuwa4z3wi`
- Then, I try to retrieve it by connect to `127.0.0.1:8000/activitylogs/vq6dtebgu3ipafvkcsuwa4z3wi/`, which leads to AttributeError.
- I also try to  retrieve it by `127.0.0.1:8000/activitylogs/23/`, and the same error shows.

**Comment by JASchilz** [comment]
I was able to replicate it and solve it. Try version 0.10.1 .

The easiest way I've found to upgrade a single package is:

```
pip uninstall django-rest-encrypted-lookup
pip install django-rest-encrypted-lookup
```

Using the `--upgrade` flag upgraded other packages that I didn't want upgraded.

You're invited to pull request in documentation for a basic urls.py, such as you outlined, into README. If you don't get to it, I will later. (:

Hope this works.
